### DISCUSSION TITLE
Can i change the credentials within the DB?

### DISCUSSION THREAD

**Discussion by Pacerino** [postalserver/postal#2762] [Help with using Postal] [answered]
I'm planning to rotate the credentials for servers with HashiCorp Vault. Does postal allows to change the key within the DB Table? I can see that they're not encrypted. Therefore i think i can just input my own string and postal will check against the table if somebody tries to authenticate themselves. Is this assumption correct? Are there any requirements for the keys eg. length?

**Comment by Pacerino** [comment, marked as answer]
I've tested this several times before and couldn't get it to work. Somehow I've managed to get it working. So the answere is yes, you can change the credentials to anything you like within the `postal.credentials` table and the `key` column. It's not encrypted nor is it required to have any specific length.

**Reply by willpower232** [reply]
If you're rotating keys on purpose, it might be useful to add a new key and wait until the credential page confirms that the old key has stopped being used to confirm it has been phased out properly.

![image](https://github.com/postalserver/postal/assets/1619102/4a7e3658-0e36-4991-989a-574d980092db)

But yes as you say, the database stores the credentials unencrypted for easy editing. We have come across some MFPs which have a password length built in to them so we had to edit Postals credential to be slightly shorter.
